.github/workflows/build.yaml

```diff
@@ -0,0 +1,478 @@
+name: Build
+on:
+  push:
+    branches:
+      - master
+  pull_request:
+    types:
+      - opened
+      - reopened
+      - synchronize
+
+jobs:
+  pre-build-updater:
+    runs-on: ubuntu-latest
+    container:
+      image: quay.io/stackrox-io/apollo-ci:scanner-test-0.3.61
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v4
+        with:
+          fetch-depth: 0
+          ref: ${{ github.event.pull_request.head.sha }}
+
+      - uses: ./.github/actions/job-preamble
+
+      - name: Cache Go dependencies
+        uses: ./.github/actions/cache-go-dependencies
+
+      - name: Build updater
+        run: make build-updater
+
+      - name: Bundle the build to preserve permissions
+        run: tar -cvzf updater-build.tgz bin/updater
+
+      - uses: actions/upload-artifact@v3
+        with:
+          name: updater-build
+          path: updater-build.tgz
+
+  pre-build-scanner:
+    runs-on: ubuntu-latest
+    container:
+      image: quay.io/stackrox-io/apollo-ci:scanner-test-0.3.61
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v4
+        with:
+          fetch-depth: 0
+          ref: ${{ github.event.pull_request.head.sha }}
+
+      - uses: ./.github/actions/job-preamble
+
+      - name: Cache Go dependencies
+        uses: ./.github/actions/cache-go-dependencies
+
+      - name: Build Scanner
+        run: make scanner-build-nodeps
```

Comment:
Honestly, I think `scanner-build-nodeps` is a bit of a misnomer. AFAICT, it still grabs the dependencies anyway. That being said, now that I look at the `deps` target, it might be worth adding that step first (since it has some other dependent targets). The main reason I used `scanner-build-nodeps` (without `deps`) was because that's how we do it in `stackrox/stackrox`

Reply:
let's keep it as-is to stay aligned with stackrox/stackrox